linera-chain/src/manager.rs

```diff
@@ -394,36 +398,45 @@ impl ChainManager {
         Ok(Outcome::Accept)
     }
 
+    /// If the validated block certificate is more recent, returns that certificate
+    /// so that the locked block can be updated.
+    pub fn validated_block_if_newer_than_locked(
```

Comment:
I'd move this method to `impl LiteCertificate` and call ` update_if_newer`. To me methods should be verbs, this is not a verb and it's not really describing what's happening.